RecoJets/JetProducers/plugins/PileupJetIdProducer.cc

```diff
@@ -64,6 +64,10 @@ PileupJetIdProducer::PileupJetIdProducer(const edm::ParameterSet& iConfig)
 // ------------------------------------------------------------------------------------------
 PileupJetIdProducer::~PileupJetIdProducer()
 {
+  for(std::vector<std::pair<std::string,PileupJetIdAlgo *> >::iterator 
+	ialgo = algos_.begin(); ialgo!=algos_.end(); ++ialgo) {
+    delete ialgo->second;
```

Comment:
it would be much better to use smart pointers (unique_ptr), they will handle deletion.
(can be done in a separate PR)